internal/auth/README.md

```diff
@@ -0,0 +1,17 @@
+# Authentication mechanism for servers
```

Comment:
This probably can go to the configauth. That is the main package, here is just the implementation.